CMakeLists.txt

```diff
@@ -48,45 +58,68 @@ if(CTPG_ENABLE_TESTS)
         tests/buffers.cpp
     )
 
-    target_link_libraries(testbin Catch2::Catch2)
-    target_include_directories(testbin PRIVATE ${PROJECT_SOURCE_DIR}/include)
-    target_compile_features(testbin PUBLIC cxx_std_17)
+    target_link_libraries(testbin PRIVATE Catch2::Catch2 ctpg::ctpg)
+    if (CTPG_WARNING_FLAGS)
+        if (MSVC)
+            target_compile_options(testbin PRIVATE /W4 /WX)
+        else ()
+            target_compile_options(testbin PRIVATE -Wall -Wextra -pedantic -Werror)
+        endif ()
+    endif ()
 
-    include(CTest)
     include(Catch)
     catch_discover_tests(testbin)
-    enable_testing()
-endif(CTPG_ENABLE_TESTS)
+endif ()
 
 # Install
-install(TARGETS ${PROJECT_NAME}
-        EXPORT ${PROJECT_NAME}_Targets
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
-
-include(CMakePackageConfigHelpers)
 
-write_basic_package_version_file("${PROJECT_NAME}ConfigVersion.cmake"
-                                 VERSION ${PROJECT_VERSION}
-                                 COMPATIBILITY SameMajorVersion)
-if(NOT INCLUDE_INSTALL_DIR)
-  set(INCLUDE_INSTALL_DIR include/${PROJECT_NAME})
-endif()
+if (NOT CTPG_ENABLE_INSTALL)
+    return()
+endif ()
+
+install(
+    TARGETS ctpg
+    EXPORT ctpg-targets
+    INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+)
+
+# Allow CTPG to be found on 32-bit systems, even if it
+# was "built" on a 64-bit system, and vice-versa.
+if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.14)
+    set(arch_opts ARCH_INDEPENDENT)
+else ()
+    set(arch_opts "")
+endif ()
+
+write_basic_package_version_file(
+    "ctpg-config-version.cmake"
+    COMPATIBILITY SameMajorVersion
+    ${arch_opts}
+)
```

Comment:
I would strongly suggest bumping the minimum version to 3.14 (or newer!) to replace this with:

```
write_basic_package_version_file(
    "ctpg-config-version.cmake"
    COMPATIBILITY SameMajorVersion
    ARCH_INDEPENDENT
)
```
There are many reasons to use a newer CMake version, but two stand out to me here:

1. GitHub Actions keeps CMake up to date by default, so it will not catch forwards-compatibility bugs without more configuration to actually use the minimum version.
2. You _really_ want your header-only library to be an architecture-independent package. It makes life a _lot_ simpler for deployment. As it stands CMake 3.12 and 3.13 cannot produce packages on x86-64 that will work on arm32, but 3.14+ can and does.

Reply:
No problem, let's do it.

Reply:
Sure. All of the changes look really nice now.
If you don't mind looking at readme.md and updating installation instructions, that would be nice.
Also look at examples in readme if the #includes are consistent with sources now.

There is the https://aur.archlinux.org/packages/ctpg-git/
@JPenuchot (the other contributor to ctpg) is maintaining it, he probably knows best how to deal with it. 

As of the Conan package, I'm not sure if it is even working, so maybe whoever is maintaining it will apply necessary changes.

Reply:
These changes have no impact on my package afaik, I'll update it to rely on `BUILD_TESTING` instead of the other variable I created as soon as the PR is merged. Thanks for the notice!